platform/darwin/src/MGLMapSnapshotter.mm

```diff
@@ -117,180 +118,198 @@ - (void)startWithQueue:(dispatch_queue_t)queue completionHandler:(MGLMapSnapshot
             }
+        }
+        
+        NSImage *logoImage = [self logoImageWithStyle:attributionInfoStyle];
+        CGSize attributionBackgroundSize = [self attributionTextSizeWithStyle:attributionInfoStyle];
+        NSImage *sourceImage = mglImage;
+        
+        CGRect logoImageRect = CGRectMake(MGLLogoImagePosition.x, MGLLogoImagePosition.y, logoImage.size.width, logoImage.size.height);
+        CGPoint attributionOrigin = CGPointMake(targetFrame.size.width - 10 - attributionBackgroundSize.width,
+                                                MGLLogoImagePosition.y + 1);
+        if (!logoImage) {
+            CGSize defaultLogoSize = [self mapboxLongStyleLogo].size;
+            logoImageRect = CGRectMake(0, MGLLogoImagePosition.y, 0, defaultLogoSize.height);
+            attributionOrigin = CGPointMake(10, attributionOrigin.y);
+        }
+        
+        CGRect attributionBackgroundFrame = CGRectMake(attributionOrigin.x,
+                                                       attributionOrigin.y,
+                                                       attributionBackgroundSize.width,
+                                                       attributionBackgroundSize.height);
+        CGPoint attributionTextPosition = CGPointMake(attributionBackgroundFrame.origin.x + 10,
+                                                      logoImageRect.origin.y + (logoImageRect.size.height / 2) - (attributionBackgroundSize.height / 2));
+        
+        
+        NSImage *compositedImage = nil;
+        NSImageRep *sourceImageRep = [sourceImage bestRepresentationForRect:targetFrame
+                                                                    context:nil
+                                                                      hints:nil];
+        compositedImage = [[NSImage alloc] initWithSize:targetSize];
+        
+        [compositedImage lockFocus];
+        
+        [sourceImageRep drawInRect: targetFrame];
+        
+        if (logoImage) {
+            [logoImage drawInRect:logoImageRect];
+        }
+        
+        NSBitmapImageRep *attributionBackground = [[NSBitmapImageRep alloc] initWithFocusedViewRect:attributionBackgroundFrame];
+        
+        CIImage *attributionBackgroundImage = [[CIImage alloc] initWithCGImage:[attributionBackground CGImage]];
+        
+        NSImage *blurredAttributionBackground = [self blurredAttributionBackground:attributionBackgroundImage];
+        
+        [blurredAttributionBackground drawInRect:attributionBackgroundFrame];
+        
+        [self drawAttributionTextWithStyle:attributionInfoStyle origin:attributionTextPosition];
+        
+        [compositedImage unlockFocus];
+        
+        
+#endif
+        
+        // Dispatch result to origin queue
+        dispatch_async(queue, ^{
+            MGLMapSnapshot* snapshot = [[MGLMapSnapshot alloc] initWithImage:compositedImage scale:self.options.scale pointForFn:pointForFn];
+            _snapshotCallback = NULL;
```

Comment:
This line is forcing callback release. I tried first calling `_snapshotCallback.reset()` but didn't work. Setting the unique pointer to NULL seems more effective.

Reply:
Is this statement necessary? `_snapshotCallback` is already nilled out above, in the completion handler passed into `-drawAttributedSnapshot:snapshotImage:pointForFn:queue:completionHandler:`. Preferably, this method would be free of side effects related to `self`.